### PR TITLE
Parse Met dates

### PR DESCRIPTION
**Pull request by jacobthill**
## Why was this change made?

- Adds macro to parse Met dates
- Calls macro in met_config.rb
- Adds tests for macro

## Was the documentation (README, API, wiki, ...) updated?

n/a

### CODE REVIEW

**Comment by jacobthill** [comment]
I'm stumped by the error message. rubocop passes locally and  the lines referenced in the circleci text don't exist in my version of the file. Any ideas what needs to be fixed?

File: lib/macros/date_parsing.rb [hunk 1]
```diff
@@ -97,6 +97,13 @@ def marc_date_range
       end
     end
 
+    # Extracts earliest & latest dates from Met record and merges into singe date range value
+    def met_date_range
+      lambda do |record, accumulator, _context|
+        first_year = record['objectBeginDate'].to_i if record['objectBeginDate'].to_s =~ /\d+/
+        last_year = record['objectEndDate'].to_i if record['objectEndDate'].to_s =~ /\d+/
+        accumulator.replace(Macros::DateParsing.year_array(first_year, last_year))
+
```
Comment (mjgiarlo) [inline lib/macros/date_parsing.rb]: Two missing `end` statements here.

Comment (jacobthill) [reply lib/macros/date_parsing.rb]: Opps, I think I deleted accidentally them when trying to merge conflicts

**Comment by jacobthill** [comment]
Not sure what to do about this error.

File: traject_configs/met_config.rb [hunk 2]
```diff
@@ -54,7 +56,7 @@
 to_field 'cho_medium', extract_json('.medium')
 to_field 'cho_date', generate_object_date, transform(&:presence)
 to_field 'cho_date', extract_json('.objectDate'), transform(&:presence)
-# to_field 'cho_date_range_norm', extract_json('.objectDate'), transform(&:presence)
+to_field 'cho_date_range_norm', extract_json('.objectDate'), transform(&:presence), met_date_range
```
Comment (ndushay) [inline traject_configs/met_config.rb]: i think this was just discussed, but I believe this should be:

```
to_field 'cho_date_range_norm', met_date_range
```

**Pull request merged**